lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallable.java

```diff
@@ -45,6 +45,9 @@
     protected final void assignArguments(List<?> args, Env e) {
         // TODO: var args
         for (int i=0; i< Math.min(args.size(),parameters.size()); i++) {
+            Class argClass = args.get(i) != null ? args.get(i).getClass() : Object.class;
+
+            e.declareVariable(argClass, parameters.get(i));
```

Comment:
This is perhaps wrong. The variable may be already declared.

Now I am thinking this is wrong. `assignArguments` is called every time the closure is used. We need to call `declareVariable` when the closure is defined.